oddbird/MetaShare
# [fix-org-failed-flow] Delete new org from SF if initial flow run fails.

## Cute animal pic
_Because everyone likes pictures of animals._

![fennec fox](https://2.bp.blogspot.com/-PnsWUDtu2k4/VGO1GZPxNsI/AAAAAAAAK1Y/7dUZr55v7LM/s1600/Fennec%2BFox_11102014_TC%2B(44).JPG)

## Link to Trello card (if applicable)
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._

https://trello.com/c/p4gpRNcC

## Description
_The commit messages say what you did; this should explain why and how._

If a scratch org creation or initial flow run fails, we were already deleting it in MS, but we were not deleting it in SF. This PR splits `create_org` from `run_flow` so that we can clean up after ourselves in the event of an error. And if that SF-deletion-because-of-a-flow-error also fails, we leave the MS org in place so that the user knows it still exists in SF, and can try deleting it again.

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally._

1. Follow steps in https://github.com/oddbird/MetaShare/pull/44#issue-332224028 to create a new org, make some changes, and capture them.
2. Create a new org (dev or QA) from that branch.

Assuming David hasn't fixed the CCI side of things yet, the creation should fail during the setup flow. You should see the error, but also see that the new scratch org has also been deleted from your SF org.

---

## Reviewer tasks:
